### ISSUE TITLE
Chromium behavior when restoring Application window

### ISSUE THREAD

**Issue by rjmedian** [dliw/fpCEF3#36]
Hello, 
I'm testing CEF 3.2526 in Lazarus 1.6 on Windows XP 32-bit. I think I encounter quite an annoying behavior of fpCEF3 which happens when restoring App Window from taskbar, after pressing '**Show Desktop**'. When restoring, and the App Window shows up, there I notice the Chromium content seems to be rendered multiple times (more than once). May be due to WM_RESIZE processing?

This behavior neither occur with any Delphi CEF3.* nor with Linux version of Lazarus 1.6 (with the same 3,2526 source). For simple reproduce, please browse using fpCEF3 to [http://www.yahoo.com](url) then click '**Show Desktop**' to hide all windows, and then try restore App Window from taskbar. You should see what I said above. 

If that's a bug, I'm really hoping this can be fixed. Thank you very much in advance.

**Comment by dliw** [comment]
Hello,
First of all I would recommend to use the latest fpCEF version (3.2743).
As I'm almost exclusively working on Linux, it's difficult for me to test on Windows - but I'll try to understand your problem.

Just to be sure: Does this problem occur with the LCLSimple or LCLCefClient example?

**Comment by rjmedian** [comment]
Thanks for the prompt reply,
Yes, it occurs on both, especially LCLCefClient which is I'm testing right now. And strangely It doesn't occur with 'minimal' example, where I think fpCEF3 in LCLCefClient should behave like that.

**Comment by dliw** [comment]
I now tested the minimal example side-by-side with LCLSimple and LCLCefClient on Windows 7 as you described. I cannot see a difference between the minimal example and the other two.
One thing I noticed though: the LCL examples use a different minimize animation, but this is easy to fix.

One thing that could happen while restoring from minimized state is, that the focus "jumps" between the LCL application and CEF more than once, which could lead to a flickering window. I didn't see any flicker, but without a slow-motion video I cannot say for sure that there is none. However I would expect this to happen on Linux rather than on Windows because of the way how CEF is embedded.

So unfortunately I cannot reproduce with the latest fpCEF version.

**Comment by rjmedian** [comment]
Never mind, thank you for spending time on this problem. 

I agree with you, a window-focusing jump will force a redraw. The only problem is the redraw seems to be issued more than once to TChromium.  Actually I 've also tried to intercept the messages received by TChromium (handling its WndProc). Yes, during the restore, there were detected: focus, lostfocus, size, and paint. Among them only (wm_)size received 3 times, and I suspect it to be the cause of the flicker. Tried to block it but unsuccessful (lack of knowledge, lol).

I already tried fpCEF version (3.2743) as you recommend, but unfortunately its dlls no longer support XP 32-bit. So, I reverted to 3.2526. In Linux 32/64 bit versions, 3.2743 is working flawlessly.

**Comment by dliw** [comment]
There are no changes between the version 3.2526 and 3.2743 related to focus handling on the fpCEF side of things, so if things changed, they were changed in CEF.

Just a quick guess:
Maybe if you filter the `PostMessage` call in [`cef3lcl.pas`](https://github.com/dliw/fpCEF3/blob/master/Component/cef3lcl.pas#L755-L761) things will improve for you.

Apart from that I cannot help you more right now - sorry. Closing this for now; feel free to reopen if you have a question or need more information.

**Comment by rjmedian** [comment]
Dear **dliw**, good news! Your quick guess was exactly the clue!! I never think to modify your component source code so I intercept the messages in the application that's why it wasn't functioning. Well, since you directed me to it, I'm feeling free now to experiment. So, instead of modifying ``TCustomChromium.WndProc``, I modified ``TCustomChromium.Resize`` ( I really suspect this one ), and below is what I did so far:
```
// fpCEF3.2526
// c3f3lcl.pas
...
...
procedure TCustomChromium.Resize;
{$IFDEF WINDOWS}
Var
  Hand: THandle;
  Rect: TRect;
{$ENDIF}
{$IFDEF LINUX}
Var
  Offset: TPoint;
{$ENDIF}
begin
  if (GetParentForm(Self).WindowState=wsMaximized) then exit; // inserted code

  inherited Resize;
...
...
end;
```
The flicker is really gone now. I don't know if the inserted code is acceptable.
I think this problem is solved for now, I'm sorry to reopen and cause you to reclose this discussion. Thank you so much again for spending time to discuss. Warm regards.
